internal/engine/evaluator.go

```diff
@@ -150,8 +150,26 @@ func (rpe *resourcePolicyEvaluator) Evaluate(ctx context.Context, tctx tracer.Co
 		}
 
 		// evaluate each rule until all actions have a result
-		for _, rule := range p.Rules {
+		for i, rule := range p.Rules {
 			rctx := sctx.StartRule(rule.Name)
+
+			if rule.Output != nil {
+				octx := rctx.StartOutput(rule.Name)
+				output, err := rpe.evalParams.evaluateStrCELExpr(rule.Output.Checked, variables, input)
+				if err != nil {
+					octx.Skipped(err, "Error evaluating output")
+				}
+
+				ruleName := rule.Name
+				if ruleName == "" {
+					ruleName = fmt.Sprintf("rule-%03d", i)
+				}
+				result.Outputs = append(result.Outputs, &enginev1.OutputEntry{
+					Src: fmt.Sprintf("%s#%s", rpe.policy.Meta.Fqn, ruleName),
```

Comment:
Let's extract this to a function in the `namer` package (`RuleFQN`?) so that if we ever have to tweak the naming convention, it can be done in a single place.

Since the rules are now actually addressable in client code, I have a feeling that we may have to normalise the rule names at some point (remove spaces etc.) If so, being able to do that in the `namer` package would be very handy.